subsys/bluetooth/host/settings.c

```diff
@@ -121,6 +121,18 @@ static int set(int argc, char **argv, char *val)
 		return 0;
 	}
 
+	if (!strcmp(argv[0], "name")) {
+		len = sizeof(bt_dev.name);
+		if (len < strlen(val)) {
+			BT_ERR("Invalid length for device name in storage: name"
+			       " will be truncated");
+		}
+
+		strncpy(bt_dev.name, val, len);
```

Comment:
The last parameter should be "len - 1" (or "sizeof(bt_dev.name) - 1") since otherwise you risk leaving it without a terminating null.

Reply:
Done.